### PR TITLE
Add CLI, fix bug

### PR DESCRIPTION
**Pull request by tsortwehttam**
1. Adds a standard Node.js CLI for this module (see README for usage).
2. Fixes a bug where files that were already `*.incremental.*` files were processed again and again, leading to ever-multiplying files being produced on each run.

Note: The `fixSeparators` call was interfering with expectations for the CLI so I added an option to disable that behavior that is only used in the CLI code path. It should not interfere with normal usage.

### CODE REVIEW

**Comment by tsortwehttam** [comment]
Hello @t3hnar just checking if someone would like to merge this PR?

**Comment by t3hnar** [comment]
@matthewtoast I'm not a maintainer of this project, however will try to poke somebody.

**Comment by hansagames** [comment]
Hi @matthewtoast, sorry for delay will try to check it as soon as possible

**Comment by geobalas** [comment]
Hello @matthewtoast, could you give more details about the `fixSeparators` issue? What didn't work correctly with cli?

File: lib/index.ts [hunk 1]
```diff
@@ -30,6 +33,12 @@ function searchBabylonFiles(root: string, currentPath: string, options: SearchOp
     readdirSync(currentPath).forEach((file: string) => {
         const filePath = join(currentPath, file);
         const babylonExtension = ".babylon";
+        const incrementalPart = ".incremental";
```
Comment (geobalas) [inline lib/index.ts]: Please move `babylonExtension` and `incrementalPart` out of this function, and filter the paths before forEach:
```typescript
readdirSync(currentPath)
    .filter((file: string) => file.indexOf(incrementalPart) !== -1)
    .forEach((file: string) => {
...
```

File: lib/index.ts [hunk 2]
```diff
@@ -17,7 +18,9 @@ export default function makeIncremental(src: string, options: OptionProps = {})
         throw new Error("No source directory provided. Please pass src in the options, e.g. /scene");
     }
 
-    const parsedSrc = fixSeparators(src);
+    const parsedSrc = (options.skipFixSeparators)
```
Comment (geobalas) [inline lib/index.ts]: Please give more information about why this is needed.

**Comment by tsortwehttam** [comment]
@geobalas Thanks for the reply!

Sorry for the delay on my end. The reason for the `skipFixSeparators` hack is that `fixSeparators()` produces "absolute" paths without leading forward-slashes. This is a problem because Node `fs.*` functions assume that path arguments without leading forward-slashes are _relative_ paths with respect to the pwd (at least on MacOS).

For example, if I pass an absolute `src` value `"/Users/matthew/foo/bar"`, `fixSeparators` converts this  to `"Users/matthew/foo/bar"`, and thus this script ultimately ends up reading from the directory `"/Users/matthew/Users/matthew/foo/bar"` (wrong). Likewise, if I pass the pwd-relative `src` value `"./foo/bar"`, then `fixSeparators` produces `"Users/matthew/foo/bar"`, which Node in turn interprets as `"/Users/matthew/Users/matthew/foo/bar"` (wrong) again. The effect is that I can't really use the script.

Since I wasn't totally sure why `fixSeparators` was needed for programmatic usage, I opted to add a hacky config option, so that code path could at least be skipped for CLI usage.

(FWIW, I think `fixSeparators` should be removed entirely from this script's logic. I think it's pretty reasonable to let the user pass any dir string that they want — and to leave it unmodified, with the assumption they know what they're doing. This also allows the script to avoid getting tangled up in platform-specific filesystem path quirks, which should fall outside of its concern.)

**Comment by geobalas** [comment]
@matthewtoast We removed the fix for the slashes, please check if it works for you now without the `skipFixSeparators` fix.

**Comment by tsortwehttam** [comment]
@geobalas Tested locally and it works now, thank you.

I added commits according to the instructions.

**Pull request merged**